posthog/temporal/ai/session_summary/activities/a1_export_session_video.py

```diff
@@ -0,0 +1,124 @@
+import uuid
+from datetime import timedelta
+
+from django.conf import settings
+from django.utils.timezone import now
+
+import structlog
+import temporalio
+from temporalio.common import RetryPolicy, WorkflowIDReusePolicy
+from temporalio.exceptions import ApplicationError
+
+from posthog.models.exported_asset import ExportedAsset
+from posthog.session_recordings.queries.session_replay_events import SessionReplayEvents
+from posthog.settings.temporal import TEMPORAL_WORKFLOW_MAX_ATTEMPTS
+from posthog.sync import database_sync_to_async
+from posthog.temporal.ai.session_summary.types.video import VideoSummarySingleSessionInputs
+from posthog.temporal.common.client import async_connect
+from posthog.temporal.exports_video.workflow import VideoExportInputs, VideoExportWorkflow
+
+from ee.hogai.session_summaries.constants import (
+    DEFAULT_VIDEO_EXPORT_MIME_TYPE,
+    EXPIRES_AFTER_DAYS,
+    MIN_SESSION_DURATION_FOR_SUMMARY_MS,
+)
+from ee.hogai.session_summaries.session.input_data import get_team
+
+logger = structlog.get_logger(__name__)
+
+
+@temporalio.activity.defn
+async def export_session_video_activity(inputs: VideoSummarySingleSessionInputs) -> int:
+    """Export full session video and return ExportedAsset ID"""
+    try:
+        # Check for existing exported asset for this session
+        existing_asset = (
+            await ExportedAsset.objects.filter(
+                team_id=inputs.team_id,
+                export_format=DEFAULT_VIDEO_EXPORT_MIME_TYPE,
+                export_context__session_recording_id=inputs.session_id,
+            )
+            .exclude(content_location__isnull=True, content__isnull=True)
+            .afirst()
+        )
+
+        if existing_asset:
+            # Check duration from existing asset's export_context
+            existing_duration_s = (
+                existing_asset.export_context.get("duration", 0) if existing_asset.export_context else 0
+            )
+            if existing_duration_s * 1000 < MIN_SESSION_DURATION_FOR_SUMMARY_MS:
+                raise ApplicationError(
+                    f"Session {inputs.session_id} video is too short for summarization "
+                    f"({existing_duration_s * 1000:.0f}ms < {MIN_SESSION_DURATION_FOR_SUMMARY_MS}ms)",
+                    non_retryable=True,
+                )
+            logger.info(
+                f"Found existing video export for session {inputs.session_id}, reusing asset {existing_asset.id}",
+                session_id=inputs.session_id,
+                asset_id=existing_asset.id,
+            )
+            return existing_asset.id
+
+        # Get session duration from metadata
+        team = await database_sync_to_async(get_team)(team_id=inputs.team_id)
+        metadata = await database_sync_to_async(SessionReplayEvents().get_metadata)(
+            session_id=inputs.session_id,
+            team=team,
+        )
+        if not metadata:
```

Comment:
Updated to log exceptions with tag in this file. I think it's worth attaching the tag to all logs to unify, so it's easier to find.